### PR TITLE
Tactician tools

### PR DESCRIPTION
**Pull request by remi-san**

### CODE REVIEW

File: tests/Tactician/CommandBusConsumerTest.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        $this->commandBus = \Mockery::mock(CommandBus::class);
+    }
+
+    /**
+     * @test
+     */
+    public function it_should_handle_the_deserialized_command()
+    {
+        $serializedCommand = json_encode([ 'foo' => 'bar' ]);
+        $command = \Mockery::mock(NamedCommand::class);
```
Comment (SelrahcD) [inline tests/Tactician/CommandBusConsumerTest.php]: I'm not sure you have to mock over here.

File: tests/Tactician/QueuePublishingMiddlewareTest.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+     */
+    public function it_should_serialize_the_command_and_publish_the_message()
+    {
+        $self = $this;
+
+        $command = \Mockery::mock(NamedCommand::class, function (NamedCommand $command) {
+            $command->shouldReceive('getCommandName')->andReturn('baz');
+        });
+
+        $this->serializer->shouldReceive('serialize')->with($command)->andReturn([ 'foo' => 'bar' ])->once();
```
Comment (SelrahcD) [inline tests/Tactician/QueuePublishingMiddlewareTest.php]: Once is not needed.

File: tests/Tactician/QueuePublishingMiddlewareTest.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        $this->publisher->shouldReceive('publish')->with(json_encode([ 'foo' => 'bar' ]), 'baz')->once();
+
+        $middleware = new QueuePublishingMiddleware($this->serializer, $this->publisher);
+        $middleware->execute(
+            $command,
+            function () use ($self) {
+                // the callable should never be called
```
Comment (SelrahcD) [inline tests/Tactician/QueuePublishingMiddlewareTest.php]: Why not ?

**Comment by SelrahcD** [comment]
Their isn't any interface for serializing Command. The only serializer we can use is something `serialize` based. Is it the only one implementation ?

**Comment by remi-san** [comment]
What do you mean ? You'd like me to add an implementation for the interface ?

File: tests/Tactician/CommandBusConsumerTest.php (hunk 1 above)
Comment (remi-san) [inline tests/Tactician/CommandBusConsumerTest.php]: True, but as I don't have an accessible class to implement it, mocking it was easier.

File: tests/Tactician/QueuePublishingMiddlewareTest.php (hunk 3 above)
Comment (remi-san) [inline tests/Tactician/QueuePublishingMiddlewareTest.php]: Yep, for consistency

File: tests/Tactician/CommandBusConsumerTest.php (hunk 1 above)
Comment (SelrahcD) [inline tests/Tactician/CommandBusConsumerTest.php]: Debatable :p

**Comment by SelrahcD** [comment]
Their is no contract to access Command data (an Interface) thus the only possible implementation of the serialize method of CommandSerializer is to use `serialize`. Is it a good idea to ask for the serialize method in CommandSerializer then ?

**Comment by remi-san** [comment]
Nope, you can serialize it a lot of different ways, with ocramius/hydrator for instance. And I don't want to force an interface on the commands so they can be serializable. But leave the option to the developers implementing it.

**Pull request merged**